src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -182,6 +177,11 @@ protected virtual string GetAssemblyName()
             return PathUtilities.GetFileName(assemblyName);
         }
 
+        protected bool IsProjectReferenceOutputAssembly(MSB.Framework.ITaskItem item)
+        {
+            return item.GetMetadata("ReferenceOutputAssembly") == "true";
```

Comment:
Would `item.GetMetadata("ReferenceSourceTarget") == "ProjectReference"` be better here?